ci/build.yml

```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.19.2
+    tag: 1.20.4
```

Comment:
This is an out of date version of go. 1.21.1 is available (https://hub.docker.com/_/golang)

Reply:
`1.21.1` doesn't seem to deploy to our environments though https://onswebsite.slack.com/archives/C024YHAF36C/p1695290998005589

Reply:
Updated to `1.21.1-bullseye`